### PR TITLE
Auth: fix check for conflict login in validation

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
This updates validation to check for conflicts in login and email, previously we only checked for email and if there was a conflict in login we did not validate that the login was a previous conflict.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/cmd/grafana-cli/commands/conflict_user_command.go [hunk 1]
```diff
@@ -256,8 +258,15 @@ func getValidConflictUsers(r *ConflictResolver, b []byte) error {
 		if err != nil {
 			return fmt.Errorf("could not parse the content of the file with error %e", err)
 		}
-		if !previouslySeenEmails[strings.ToLower(newUser.Email)] {
-			return fmt.Errorf("not valid email: %s, email not in previous conflicts seen", newUser.Email)
+		if newUser.ConflictEmail != "" {
```
Comment (linoman) [inline pkg/cmd/grafana-cli/commands/conflict_user_command.go:261]: Is there a specific reason to not join these 2 conditionals with `AND`?

**Review by linoman — APPROVED** [review]
LGTM! Great job!

I don't know how this works for the CLI command, but do we need tests for the CLI changes?

File: pkg/cmd/grafana-cli/commands/conflict_user_command.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/cmd/grafana-cli/commands/conflict_user_command.go:261]: no real reason. Could refactor into a function call to make it clear what we are after

Comment (eleijonmarck) [reply pkg/cmd/grafana-cli/commands/conflict_user_command.go]: it's a situation of AND/OR for conflicts here.

I will merge this and see if we want to refactor that later.

Made into one if statement

**Pull request merged**